src/err/mod.rs

```diff
@@ -34,6 +36,9 @@ pub enum Error {
 	#[error("There was a problem connecting with the storage engine")]
 	InvalidStorage,
 
+	#[error("There was a problem parsing the header {0}: {1}")]
+	InvalidHeader(HeaderName, TypedHeaderRejection),
```

Comment:
`axum` hasn't reached 1.0 yet. Making it a public dependency like this in a stable crate will be problematic.